### PR TITLE
Pod controller interface

### PR DESCRIPTION
**Pull request by e-sumin**
## Change Overview

Introduce interfaces for POD manipulation.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Review by denisvmedia — APPROVED** [review]

**Review by viveksinghggits — APPROVED** [review]

File: pkg/kube/pod_command_executor.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2019 The Kanister Authors.
```
Comment (pavannd1) [inline pkg/kube/pod_command_executor.go:1]:
```suggestion
// Copyright 2023 The Kanister Authors.
```

File: pkg/kube/pod_controller.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 The Kanister Authors.
```
Comment (pavannd1) [inline pkg/kube/pod_controller.go:1]:
```suggestion
// Copyright 2023 The Kanister Authors.
```

File: pkg/kube/pod_file_writer.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 The Kanister Authors.
```
Comment (pavannd1) [inline pkg/kube/pod_file_writer.go:1]:
```suggestion
// Copyright 2023 The Kanister Authors.
```

File: pkg/kube/pod_file_writer.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	"io"
+)
+
+// PodFileRemover provides the mechanism to remove written file from the pod.
+type PodFileRemover interface {
+	Remove(ctx context.Context) error
+	Path() string
+}
+
+// PodFileWriter allows us to write file to POD.
```
Comment (pavannd1) [inline pkg/kube/pod_file_writer.go:28]:
```suggestion
// PodFileWriter allows us to write files to the pod.
```
[nit] s/POD/pod for consistency

File: pkg/kube/pod_writer.go [hunk 5]
```diff
@@ -25,24 +25,34 @@ import (
 	"github.com/kanisterio/kanister/pkg/format"
 )
 
-// PodWriter specifies Kubernetes Client and the other params needed for writing content to a file
-type PodWriter struct {
+// PodWriter specifies interface needed for manipulating with file in pod
```
Comment (pavannd1) [inline pkg/kube/pod_writer.go:28]:
```suggestion
// PodWriter specifies interface needed for manipulating files in a pod
```
[nit]

**Review by pavannd1 — APPROVED** [review]

File: pkg/kube/pod_controller.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+package kube
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
```
Comment (PrasadG193) [inline pkg/kube/pod_controller.go:23]: [nit] Add comment

**Pull request merged**